app/src/main/java/org/simple/clinic/recentpatient/RecentPatientRelativeTimeStampGenerator.kt

```diff
@@ -0,0 +1,50 @@
+package org.simple.clinic.recentpatient
+
+import android.content.Context
+import org.simple.clinic.R
+import org.threeten.bp.LocalDate
+import org.threeten.bp.ZoneOffset
+import org.threeten.bp.format.DateTimeFormatter
+import java.util.Locale
+import javax.inject.Inject
+
+class RecentPatientRelativeTimeStampGenerator @Inject constructor() {
+
+  fun generate(today: LocalDate, date: LocalDate): RelativeTimestamp {
+    val yesterday = today.minusDays(1)
+
+    return when (date) {
+      today -> Today
+      yesterday -> Yesterday
+      else -> OlderThanTwoDays(date)
+    }
+  }
+
+  fun generate(date: LocalDate): RelativeTimestamp {
+    val today = LocalDate.now(ZoneOffset.UTC)
+    return generate(today, date)
+  }
+}
+
+sealed class RelativeTimestamp {
+  abstract fun displayText(context: Context): String
+}
+
+object Today : RelativeTimestamp() {
+  override fun displayText(context: Context): String =
+      context.getString(R.string.timestamp_today)
+}
+
+object Yesterday : RelativeTimestamp() {
+  override fun displayText(context: Context): String =
+      context.getString(R.string.timestamp_yesterday)
+}
+
+data class OlderThanTwoDays(private val date: LocalDate) : RelativeTimestamp() {
+  override fun displayText(context: Context): String =
+      timestampFormatter.format(date)
+
+  companion object {
+    private val timestampFormatter = DateTimeFormatter.ofPattern("d MMMM, yyyy", Locale.ENGLISH)
```

Comment:
Can you inject this via the `DateFormatterModule`?

Reply:
done